bin/tools/Templates/FrontendUnitTest.js

```diff
@@ -1013,6 +1058,78 @@ export class FrontendUnitTest extends AbstractTemplate {
    * @param {String} pathToClass
    * @returns {Boolean}
    */
+  static hasImports(pathToClass) {
+
+    let fileContentString = fs.readFileSync(pathToClass, 'utf8');
+
+    let re = /import.*["']([a-z0-9-\.\/]+)['"]/gmi;
+
+    if (!re.test(fileContentString)) {
+      return false;
+    } else {
+      let imports = fileContentString.match(re);
+
+      for (let item of imports) {
+        if (item.indexOf('/name') === -1) {
+          return true;
+        }
+      }
+      return false;
+    }
+
+
+  }
+
+  /**
+   * @param {String} pathToClass
+   * @returns {String[]}
+   */
+  static getInjectedDepsForDirectiveCtrl(pathToClass) {
+
+    let fileContentString = fs.readFileSync(pathToClass, 'utf8');
+
+    let re = /controller\(.*\[([\s\S]*)\(.*\{.*/mi;
+
+    if (re.test(fileContentString)) {
+      let injectedDepsString = fileContentString.match(re)[1].replace(/[\s]/mg, '').replace(/"|'|`/mg, '');
+      return injectedDepsString.split(',').filter((element) => {
+        return element;
+      });
+    }
+
+    return [];
+  }
+
+  /**
+   * @param {String[]} depsArray
+   * @returns {String[]}
+   */
+  static fetchServices(depsArray) {
+    return depsArray.filter((element) => {
+      return /service/i.test(element);
+    });
+  }
+
+  /**
+   * @param {String[]} depsArray
+   * @returns {Boolean}
+   */
+  static containsProvider(depsArray) {
+    return /provider|notification|msAuthentication/gmi.test(depsArray.join(','));
+  }
+
+  /**
+   * @param {String[]} depsArray
+   * @returns {Boolean}
+   */
+  static containsService(depsArray) {
```

Comment:
Angular doesn't restrict to suffix services with **Service**.
I think it would be better to look at angular defined service instead of checking if there is **suffix** term in deps array